### PR TITLE
LPS-197617 Refactor interface LayoutStructureItemMapper

### PR DESCRIPTION
**Pull request by Valmir-unicap**
Related: https://github.com/liferay-core-infra/liferay-portal/pull/5631#issuecomment-1738236334

### CODE REVIEW

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:relevant

**Comment by Valmir-unicap** [comment]
ci:test:sf

**Comment by Valmir-unicap** [comment]
ci:test:sf

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/CollectionItemLayoutStructureItemMapper.java [hunk 1]
```diff
@@ -20,7 +20,10 @@
 /**
  * @author Jürgen Kappler
  */
-@Component(service = LayoutStructureItemMapper.class)
+@Component(
+	property = "layout.structure.item.mapper.type=collectionItem",
+	service = LayoutStructureItemMapper.class
```
Comment (rafaprax) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/CollectionItemLayoutStructureItemMapper.java:25]: use kebab-case instead.

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/ContainerLayoutStructureItemMapper.java [hunk 2]
```diff
@@ -39,7 +39,10 @@
 /**
  * @author Jürgen Kappler
  */
-@Component(service = LayoutStructureItemMapper.class)
+@Component(
+	property = "layout.structure.item.mapper.type=section",
+	service = LayoutStructureItemMapper.class
```
Comment (rafaprax) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/ContainerLayoutStructureItemMapper.java:44]: Why not `container`?

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/util/PageElementUtil.java [hunk 3]
```diff
@@ -92,14 +90,9 @@ private static PageElement _toPageElement(
...
-		BundleContext bundleContext = bundle.getBundleContext();
-
 		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
-			bundleContext, LayoutStructureItemMapper.class, null,
-			ServiceReferenceMapperFactory.create(
-				bundleContext,
-				(layoutStructureItemMapper, emitter) -> emitter.emit(
-					layoutStructureItemMapper.getClassName())));
+			bundle.getBundleContext(), LayoutStructureItemMapper.class,
+			"layout.structure.item.mapper.type");
```
Comment (rafaprax) [inline modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/util/PageElementUtil.java:95]: by the way, this code is broken now. In line 78, the get is made by passing the class name.

Comment (rafaprax) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/util/PageElementUtil.java:95]: please fix that, the property value should be the classname

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/ContainerLayoutStructureItemMapper.java (hunk 2 above)
Comment (Valmir-unicap) [reply modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/dto/v1_0/mapper/ContainerLayoutStructureItemMapper.java:44]: I based it on the value of the class, type = Type.SECTION, on line 116

But sorry, I'll fix that now! :)

**Pull request closed without merging**